src/Database/Type/DateTimeType.php

```diff
@@ -165,7 +169,7 @@ public function marshal($value)
                 $date = new $class($value);
                 $compare = true;
             }
-            if ($compare && $date && $date->format($this->_format) !== $value) {
+            if ($compare && $date && !$this->_compare($date, $value)) {
```

Comment:
@dereuromark Changing string variable to array is not backward compatible. If someone uses a sub-class using `_format` property directly, the application would break. So this change must be done in 3.next.